other-examples/binarize_fsl.py

```diff
@@ -9,25 +9,39 @@
 import os
 import subprocess
 from hdfs import Config
+
 import sys
+from  common_functions import directory_exists,tmpfs_directories, hdfs_status
 
-def copy_to_hdfs(x, output, client):
+def copy_to_hdfs(x, output, hdfs, tmpfs_path, client):
+    # Content not being written to the file, apparently!
     filename = x[0].split('/')[-1]
-    client.upload(output,os.path.join('/run/user/1000',filename), overwrite=True)
-    subprocess.check_output("rm {0}".format(os.path.join('/run/user/1000',filename)), shell=True)
+    if not(directory_exists(tmpfs_path)):
+        tmpfs_path = tmpfs_directories()
+    if tmpfs_path != "":
+        client.upload(output, os.path.join(tmpfs_path, filename), overwrite=True)
+        subprocess.check_output("rm {0}".format(os.path.join(tmpfs_path,filename)), shell=True)
+    else:
+        print("ERROR# Temporary file system not found while uploading")
     return (x[0], 0)
 
-def save_nifti(x):
+
+def save_nifti(x, tmpfs_path):
     filename = x[0].split('/')[-1]
     im = nib.Nifti1Image(x[1][1], x[1][0])
-    nib.save(im, os.path.join('/run/user/1000', filename))
+    nib.save(im, os.path.join(tmpfs_path, filename))
     return filename
 
-def binarize(x, threshold):
-    filename = save_nifti(x)
-    run_command = "sudo docker run -v /run/user/1000:/binarized_tmp --rm fsl-docker-img bin/fsl_bin.sh {0} {1}".format(filename, threshold)
 
-    subprocess.check_output(run_command, shell=True, stderr=subprocess.STDOUT)
+def binarize(x, threshold, tmpfs_path):
+    if not(directory_exists(tmpfs_path)):
```

Comment:
tmpfs_path should be optional. If it is None, you should figure it out using tmpfs_directories.